### DISCUSSION TITLE
Huntsman matetrack results

### DISCUSSION THREAD

**Discussion by dav1312** [joergoster/Stockfish-old#9] [Show and tell]
Results of https://github.com/vondele/matetrack using Huntsman 1 and [Stockfish Dev](https://github.com/official-stockfish/Stockfish/commit/e31f97e3baa52042fe60d6f4eeb50fe0d9e61013)

| 1 000 000 nodes | Stockfish | Huntsman |
|-|-|-|
| Total Fens | 6566 | 6566 |
| Found Mates | 3160 | 5179 |
| Best Mates | 2231 | 3811 |

**Comment by joergoster** [comment]
Looks like a good start.
Thank you for running this!

**Comment by joergoster** [comment]
Here the results of current huntsman-dev with 1000, 10,000 and 100,000 nodes.

```
Using ./huntsman with --nodes 1000
Total fens:    6560
Found mates:   46
Best mates:    35
Complete PVs:  46/46 (100.0%)
```

```
Using ./huntsman with --nodes 10000
Total fens:    6560
Found mates:   435
Best mates:    356
Complete PVs:  435/435 (100.0%)
```

```
Using ./huntsman with --nodes 100000
Total fens:    6560
Found mates:   2485
Best mates:    1792
Complete PVs:  2485/2485 (100.0%)
```
Tests were run with current matetrack-dev.

**Reply by dav1312** [reply]
```
huntsman1.exe with --nodes 1000
Total fens:    6560
Found mates:   46
Best mates:    36
Complete PVs:  41/46 (89.1%)
Wrong mates:   3
```
```
huntsman1.exe with --nodes 10000
Total fens:    6560
Found mates:   615
Best mates:    458
Complete PVs:  520/615 (84.6%)
```
```
huntsman1.exe with --nodes 100000
Total fens:    6560
Found mates:   2941
Best mates:    2064
Complete PVs:  2752/2941 (93.6%)
```
_Not sure why the values for 1M are this much different than before, the file didn't change that much_
```
huntsman1.exe with --nodes 1000000
Total fens:    6560
Found mates:   5240
Best mates:    3868
Complete PVs:  5040/5240 (96.2%)
Better mates:  1
Wrong mates:   1
```

**Comment by joergoster** [comment]
@dav1312 Which version did you use? Differences are probably to the more frequent check of check_time(). Variable `callsCnt` has been changed from 1024 to 512.

**Reply by dav1312** [reply]
huntsman1(.exe), the same as in my first comment

**Reply by joergoster** [reply]
I see. Maybe some positions were changed since the last time or simply some timing difference, idk.

**Comment by joergoster** [comment]
I've put together a small collection of mate positions from mate1 to mate4 from the matetrack repository.
Here is a comparison of **Stockfish**, **Matefish** (AB search) and **Huntsman**.

`python3 matecheck.py --engine ./matefish --epdFile mates1to4.epd --mate 0 --concurrency 1 --showAllIssues --bench`

> Loaded 111 FENs, with max(abs(bm)) = 4.
> 
> Using ./matefish on mates1to4.epd with --mate 0
> Engine ID:     Matefish 050824 64 BMI2
> Total FENs:    111
> Found mates:   111
> Best mates:    111
> 
> Best mate statistics:
> abs(bm) = 1 - mates: 4, nodes (min avg max): 1 1 1, depth (min avg max): 1 1 1
> abs(bm) = 2 - mates: 17, nodes (min avg max): 5 141 558, depth (min avg max): 3 3 3
> abs(bm) = 3 - mates: 23, nodes (min avg max): 8 1584 17414, depth (min avg max): 3 3 5
> abs(bm) = 4 - mates: 67, nodes (min avg max): 18 30032 293991, depth (min avg max): 3 5 7
> All best mates: 111, nodes (min avg max): 1 18477 293991, depth (min avg max): 1 4 7
> 
> ===========================
> Total time (ms) : 1106
> Nodes searched  : 2050975
> Nodes/second    : 1854408


`python3 matecheck.py --engine ./huntsman --epdFile mates1to4.epd --mate 0 --time 10 --concurrency 1 --showAllIssues --bench`

> Loaded 111 FENs, with max(abs(bm)) = 4.
> 
> Using ./huntsman on mates1to4.epd with --time 10.0 --mate 0
> Engine ID:     The Huntsman 240624
> Total FENs:    111
> Found mates:   111
> Best mates:    111
> 
> Best mate statistics:
> abs(bm) = 1 - mates: 4, nodes (min avg max): 1 18 64, depth (min avg max): 1 1 1
> abs(bm) = 2 - mates: 17, nodes (min avg max): 20 1403 6412, depth (min avg max): 1 7 15
> abs(bm) = 3 - mates: 23, nodes (min avg max): 150 31557 416431, depth (min avg max): 3 11 20
> abs(bm) = 4 - mates: 67, nodes (min avg max): 615 170713 2335734, depth (min avg max): 6 15 26
> All best mates: 111, nodes (min avg max): 1 109798 2335734, depth (min avg max): 1 12 26
> 
> ===========================
> Total time (ms) : 6646
> Nodes searched  : 12187543
> Nodes/second    : 1833816


`python3 matecheck.py --engine ./stockfish --epdFile mates1to4.epd --mate 0 --time 10 --concurrency 1 --showAllIssues --bench`

> Loaded 111 FENs, with max(abs(bm)) = 4.
> 
> Using ./stockfish on mates1to4.epd with --time 10.0 --mate 0
> Engine ID:     Stockfish dev-20240808-nogit
> Total FENs:    111
> Found mates:   102
> Best mates:    102
> 
> Best mate statistics:
> abs(bm) = 1 - mates: 4, nodes (min avg max): 4 66 204, depth (min avg max): 1 1 1
> abs(bm) = 2 - mates: 17, nodes (min avg max): 111 476957 8020819, depth (min avg max): 1 15 41
> abs(bm) = 3 - mates: 22, nodes (min avg max): 148 370790 5990415, depth (min avg max): 2 22 40
> abs(bm) = 4 - mates: 59, nodes (min avg max): 1365 1297916 11952449, depth (min avg max): 10 27 45
> All best mates: 102, nodes (min avg max): 4 910225 11952449, depth (min avg max): 1 23 45
> 
> ===========================
> Total time (ms) : 147692
> Nodes searched  : 252069796
> Nodes/second    : 1706726
